spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/request/Response.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.client.request;
+
+import org.candlepin.spec.bootstrap.client.ApiClient;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+
+
+
+/**
+ * An encapsulated response generated from a request. The response contains the HTTP response code,
+ * response headers, and the body in binary format. This class also provides utility methods for
+ * translating the body to various objects.
+ */
+public class Response {
+    private final int code;
+    private final byte[] body;
+    private final Map<String, List<String>> headers;
+
+    /**
+     * Create a new Response from the given OkHttp Response object.
+     *
+     * @param response
+     *  the OkHttp response object from which to build this response instance
+     *
+     * @throws IllegalArgumentException
+     *  if the inbound response object is null
+     *
+     * @throws ResponseProcessingException
+     *  if the response code, body, or headers cannot be read or processed
+     */
+    public Response(okhttp3.Response response) {
+        if (response == null) {
+            throw new IllegalArgumentException("response is null");
+        }
+
+        try {
+            this.code = response.code();
+            this.body = response.body().bytes();
```

Comment:
Is it possible for the `response.body()` to be null?

Reply:
Hrmm... maybe. That's largely copied from what the generated code does, and the [docs](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response/body/) suggest it shouldn't be null in the cases we use it.

We could throw a check in there just to be safe, though.

Reply:
Done